### ISSUE TITLE
Formatting snippets on hover

### ISSUE THREAD

**Issue by titoBouzout** [sublimelsp/LSP-typescript#243]
I'm not really sure how this could work, but would be great if the code snippets shown on hover could have a bit better formatting. Some typescript messages are very hard to parse, well even with formatting xD, but formatted would make the code snippets digestible. Thanks

**Comment by titoBouzout** [comment]
wrong repo, I mean to post in LSP

**Comment by rchl** [comment]
More correct than LSP.

**Comment by rchl** [comment]
There is a project that could be integrated with the server - https://github.com/yoavbls/pretty-ts-errors

Though it is currently not quite compatible with LSP as it formats messages to HTML rather than markdown (which LSP expects). Someone [made some effort](https://github.com/yoavbls/pretty-ts-errors/issues/21#issuecomment-1925314596) to create version that produces markdown output but it's not that good for now.

**Comment by rchl** [comment]
And note that that project is about formatting typescript error messages and nothing else (you mention code snippets but I guess you meant error messages).

**Comment by titoBouzout** [comment]
> There is a project that could be integrated with the server - https://github.com/yoavbls/pretty-ts-errors
> 
> Though it is currently not quite compatible with LSP as it formats messages to HTML rather than markdown (which LSP expects). Someone [made some effort](https://github.com/yoavbls/pretty-ts-errors/issues/21#issuecomment-1925314596) to create version that produces markdown output but it's not that good for now.

Yes, something like that project would be perfect. Wish ST supported JavaScript extensions..
